### ISSUE TITLE
failure 0.1.2 fails to compile on 1.27 nightly compilers from before the dyn Trait stabalization

### ISSUE THREAD

**Issue by dgriffen** [rust-lang-nursery/failure#235]
This was reported originally on IRC. If you are using a nightly compiler that was marked as 1.27 but before dyn trait was actually stabalized compilation is now broken. Some users may be stuck on an older version of nightly and now they can't compile any crates with failure as a dep.

**Comment by Isan-Rivkin** [comment]
Same problem here

**Comment by BurntSushi** [comment]
I don't think we should be supporting arbitrary versions of the nightly compiler.

**Comment by dgriffen** [comment]
Only supporting the latest nightly isn't really feasible either since many people pin their nightly versions because nightly only libraries break often.

**Comment by otavio** [comment]
@dgriffen I understand but when someone decides to use a nightly compiler, it comes with a compromise of a moving target. 

I think we must be concerned about current nightly releases otherwise the maintenance burden is giant. Users are free to pin failure releases for a compatible version of their pinned compiler if needed but ask it to be supported seems too much.

**Comment by mitsuhiko** [comment]
The feature detection for dyn is here: https://github.com/rust-lang-nursery/failure/blob/94158d21235f13635f0b0143c40db6b594fee723/failure_derive/build.rs#L6-L39

If someone has suggestions for how to make this less crude I can push out a release. My understanding is that we can only do this based on versions detection though.

**Comment by mitsuhiko** [comment]
I guess an alternative could be to try to test compile an almost empty rust file.
